app_dart/lib/src/foundation/utils.dart

```diff
@@ -186,7 +186,14 @@ Future<String> githubFileContent(
           ),
       retryIf: (Exception e) => e is HttpException || e is NotFoundException,
     );
-  } catch (e) {
+  } on Exception catch (e) {
+    // A logical error (i.e. something like an ArgumentError) should not be
+    // used as a retry signal. For example, 'TestFailure' (package:test) is an
+    // exception but is not recoverable.
```

Comment:
this was one of my "oh, my passing test were actually failing but the error expect() was swallowed" for me.